addons/mrp/models/mrp_production.py

```diff
@@ -2595,7 +2596,7 @@ def _check_sn_uniqueness(self):
                 continue
             for move_line in move.move_line_ids:
                 if not move_line.picked or float_is_zero(move_line.quantity, precision_rounding=move_line.product_uom_id.rounding) or\
-                        not move_line.lot_id:
+                        not move_line.lot_id or move_line.lot_id == self.lot_producing_id:
```

Comment:
You need to handle that in the `cancelled_sml_groups`, as move lines that comes from a re-prod cancel the consumption as well.

The issue with leaving it there is that in the following case :
- Use the SN in a manufacturing order (whatever the out product)
- Now create a SN that both uses that SN and produces it.
It ignores all checks then outputs a Validation Error because you tried to have a quantity of 2 in the production quant. And that Validation Error could be confusing on to what actually happens, rather than using the correct one here.